frappe/api.py

```diff
@@ -245,5 +245,6 @@ def validate_api_key_secret(api_key, api_secret, frappe_authorization_source=Non
 			)
 		else:
 			user = frappe.db.get_value(doctype, doc, 'user')
-		frappe.set_user(user)
+		if not frappe.get_user():
```

Comment:
this will always return some value since it works for when the user is set to "Guest" as well. Maybe you can here instead check for whether the `get_user().name` is set to the current user that is trying to login?

Reply:
@Thunderbottom something like this?
```suggestion
		if not frappe.local.login_manager.user in ('', 'Guest')::
```

Reply:
or something like where you can check if the current user isn't the same as the one that is trying to login?

Reply:
Well, that's exactly what we want...
In our case the logged in user and the apiKey user may be different and in this code the logged in one gets overwritten.
With this change the goal is, to not overwrite the logged in user.

Reply:
then i think checking whether the logged in user is not guest would solve your issue.

Reply:
So my suggestion above